File: MaximumAscendingSubArraySum/Leetcode1800.java
package MaximumAscendingSubArraySum;

class Leetcode1800 {
    public int maxAscendingSum(int[] nums) {
        int prev = nums[0];
        int maxSum = nums[0];
        int currSum = nums[0];

        for(int i = 1; i < nums.length; i++) {
            if(nums[i] > prev) {
                currSum += nums[i];
            } else {
                currSum = nums[i];
            }
            if(currSum > maxSum) {
                maxSum = currSum;
            }
            prev = nums[i];
        }
        return maxSum;
    }
}

File: MaximumAscendingSubArraySum/Leetcode1800.py
class Leetcode1800:
    def maxAscendingSum(self, nums: List[int]) -> int:
        prev = nums[0]
        currSum = nums[0]
        maxSum = nums[0]
        for num in nums[1:]:
            if num > prev:
                currSum = currSum + num
            else:
                currSum = num
            
            if currSum > maxSum:
                maxSum = currSum
            
            prev = num

        return maxSum